view/frontend/web/js/action/place-order.js

```diff
@@ -58,43 +57,54 @@ define(
                 serviceUrl,
                 JSON.stringify(payload)
             ).done(
                 function (response) {
                     // Redirect URL
                     if (response === true) {
                         checkoutSessionUpdateAction(function (redirectUrl) {
-                            customerData.invalidate(['cart']);
-                            customerData.set('checkout-data', {
-                                'selectedShippingAddress': null,
-                                'shippingAddressFromData': null,
-                                'newCustomerShippingAddress': null,
-                                'selectedShippingRate': null,
-                                'selectedPaymentMethod': null,
-                                'selectedBillingAddress': null,
-                                'billingAddressFromData': null,
-                                'newCustomerBillingAddress': null
-                            });
-                            amazonStorage.clearAmazonCheckout();
-                            window.location.replace(redirectUrl);
+                            orderModel.place(
+                                amazonStorage.getCheckoutSessionId()
+                            ).done(
+                                function (response) {
+                                    if(response['success'] === false) {
+                                        console.log('Failed to place website order: ');
+                                        handleResponseError(response);
+                                        return;
+                                    }
+                                    customerData.invalidate(['cart']);
+                                    customerData.set('checkout-data', {
+                                        'selectedShippingAddress': null,
+                                        'shippingAddressFromData': null,
+                                        'newCustomerShippingAddress': null,
+                                        'selectedShippingRate': null,
+                                        'selectedPaymentMethod': null,
+                                        'selectedBillingAddress': null,
+                                        'billingAddressFromData': null,
+                                        'newCustomerBillingAddress': null
+                                    });
+                                    amazonStorage.clearAmazonCheckout();
+                                    window.location.replace(redirectUrl);
+                                }
+                            )
                         });
                     } else {
-                        fullScreenLoader.stopLoader(true);
                         console.log('Invalid Amazon RedirectUrl:');
-                        console.log(response);
-                        errorProcessor.process(response);
+                        handleResponseError(response);
```

Comment:
Doesn't seem like the customer gets notified of anything here.

Reply:
handleResponseError uses `vendor/magento/module-checkout/view/frontend/web/js/model/error-processor.js` to notify user, stop loader, and redirect if necessary.

Reply:
Got it.